### PR TITLE
Add Kusama transactions

### PR DESCRIPTION
**Pull request by hewigovens**
transfer only, we can add other types later

### CODE REVIEW

File: platform/polkadot/api.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	txs := make([]blockatlas.Tx, 0)
+	for _, srcTx := range s {
+		tx := p.NormalizeTransfer(&srcTx)
+		txs = append(txs, tx)
+	}
+
+	return txs, nil
+}
+
+func (p *Platform) NormalizeTransfer(srcTx *Transfer) blockatlas.Tx {
```
Comment (vikmeup) [inline platform/polkadot/api.go]: Can you call it NormalizeTransaction?

File: platform/polkadot/client.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+	var data SubscanResponseData
+	err = json.Unmarshal(bytes, &data)
+	if err != nil {
+		return nil, err
+	}
+	return data.Extrinsics, nil
+}
+
+func (c *Client) GetCurrentBlock() (int64, error) {
```
Comment (vikmeup) [inline platform/polkadot/client.go]: If this implemented here, can we add block parsing?

File: platform/polkadot/formatter.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package polkadot
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+)
+
+func ParseAmount(s string, decimals uint) blockatlas.Amount {
```
Comment (vikmeup) [inline platform/polkadot/formatter.go]: What does this do? We already have decoders for this

If it’s unique - we should have tests

File: platform/polkadot/model.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package polkadot
+
+type Metadata struct {
+	BlockNum    string `json:"blockNum"`
+	BlockTime   string `json:"blockTime"`
+	ImplName    string `json:"implName"`
+	NetworkNode string `json:"networkNode"`
```
Comment (vikmeup) [inline platform/polkadot/model.go]: Do you need all of the data? Can we only keep what’s needed? 


Can name BlockNum as BlockNumber? Same for others?

File: platform/polkadot/api.go (hunk 1 above)
Comment (hewigovens) [reply platform/polkadot/api.go]: they're called `Transfer` and `Extrinsic` and from two apis, `NormalizeExtrinsic` also added

File: platform/polkadot/client.go (hunk 2 above)
Comment (hewigovens) [reply platform/polkadot/client.go]: Added, will test it later

File: platform/polkadot/formatter.go (hunk 3 above)
Comment (hewigovens) [reply platform/polkadot/formatter.go]: tests added

File: platform/polkadot/model.go (hunk 4 above)
Comment (hewigovens) [reply platform/polkadot/model.go]: simplified

File: platform/polkadot/api.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+		Date:     int64(srcTx.Timestamp),
+		Block:    srcTx.BlockNumber,
+		Status:   status,
+		Sequence: srcTx.Nonce,
+	}
+
+	decimals := p.Coin().Decimals
+	if srcTx.CallModule == ModuleBalances &&
+		srcTx.CallModuleFunction == ModuleFunctionTransfer {
+		result.From = srcTx.AccountId // FIXME
```
Comment (Pantani) [inline platform/polkadot/api.go]: can you add more description about this FIXME?

File: platform/polkadot/client.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	blockatlas.Request
+}
+
+func (c *Client) GetTransfersOfAddress(address string) ([]Transfer, error) {
+	var res SubscanResponse
+	err := c.Post(&res, "/scan/transfers", TransfersRequest{Address: address, Row: blockatlas.TxPerPage})
+	if err != nil {
+		return nil, err
+	}
+	return res.Data.Transfers, nil
```
Comment (Pantani) [inline platform/polkadot/client.go]: the `Data` object can be nil?

File: platform/polkadot/client.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+func (c *Client) GetExtrinsicsOfAddress(address string) ([]Extrinsic, error) {
+	var res SubscanResponse
+	err := c.Post(&res, "/scan/extrinsics", TransfersRequest{Address: address, Row: blockatlas.TxPerPage})
+	if err != nil {
+		return nil, err
+	}
+	return res.Data.Extrinsics, nil
```
Comment (Pantani) [inline platform/polkadot/client.go]: the `Data` object can be nil?

File: platform/polkadot/api.go (hunk 5 above)
Comment (hewigovens) [reply platform/polkadot/api.go]: Fixed

File: platform/polkadot/client.go (hunk 6 above)
Comment (hewigovens) [reply platform/polkadot/client.go]: usually not

File: platform/polkadot/api.go [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+	decimals := p.Coin().Decimals
+	if srcTx.CallModule == ModuleBalances &&
+		srcTx.CallModuleFunction == ModuleFunctionTransfer {
+		result.From = srcTx.AccountId
+		result.To = p.NormalizeAddress(datas[0].ValueRaw)
+		result.Fee = blockatlas.Amount(FeeTransfer)
+		result.Meta = blockatlas.Transfer{
+			Value:    blockatlas.Amount(fmt.Sprintf("%.0f", datas[1].Value.(float64))),
+			Symbol:   p.Coin().Symbol,
+			Decimals: decimals,
```
[the remark this answers is not in the document]
Comment (hewigovens) [reply platform/polkadot/api.go]: might be used in other staking txs later

File: platform/polkadot/api.go [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+		}
+	} else {
+		// not supported yet
+		return nil
+	}
+	return &result
+}
+
+func (p *Platform) NormalizeAddress(valueRaw string) string {
```
[the remark this answers is not in the document]
Comment (hewigovens) [reply platform/polkadot/api.go]: added

**Pull request merged**